File: Problems/NepsAcademy/e19.py

n = int(input())
resp = 0

sapatos = [[0] * 31 for _ in range(31)]

for i in range(n):
    m, l = input().split()
    m = int(m)

    m -= 30
    if l == "E":
        sapatos[m][0] += 1
        
    else:
        sapatos[m][1] += 1
    
for i in range(31):
    resp += min(sapatos[i][0], sapatos[i][1])

print(resp)

File: Problems/NepsAcademy/cpp/e19.cpp


#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n, i, numero, resp = 0;
    char lado;
    vector<pair<int, int>> sapatos(31);

    cin >> n;

    for(i = 0; i < n; i++){
        cin >> numero >> lado;

        numero -= 30;

        if (lado == 'E'){
            sapatos[numero].first += 1;
        } else {
            sapatos[numero].second += 1;
        }
    }

    
    for(i = 0; i < 31; i++){
        resp += min(sapatos[i].first, sapatos[i].second);
   
    }

    cout << resp << "\n";
    
}